packages/typespec-azure-portal-core/src/lib.ts

```diff
@@ -0,0 +1,37 @@
+import { createTypeSpecLibrary } from "@typespec/compiler";
+
+export const $lib = createTypeSpecLibrary({
+  name: "@azure-tools/typespec-azure-portal-core",
+  diagnostics: {
+    fileNotFound: {
+      severity: "error",
+      messages: {
+        browseargQuery: `cannot find @browse file argQuery`,
+        aboutIcon: `cannot find @about file icon`,
+      },
+    },
+    invalidUsageDecorator: {
+      severity: "error",
+      messages: {
+        browse: `@browse decorator can be only applied to trackedResource and proxyResource`,
+        about: `@about decorator can be only applied to trackedResource and proxyResource`,
+        marketplaceOffer: `@marketPlaceOffer decorator can be only applied to trackedResource and proxyResource`,
+      },
+    },
```

Comment:
I suggest making the diagnostic names very specific and parameterizing, like this:

```suggestion
   "not-a-resource": {
      severity: "error",
      messages: {
        default: paramMessage`@${"decoratorName"} can only be applied to TrackedResource and ProxyResource models`
      },
    },
```

This way you don't have to repeat the message, and the diagnostic name (which appears in error messages) alone communicates what's gone wrong.

Reply:
fixed :) thank you!!